### ISSUE TITLE
The experiment loop lacks metacognition: no stall detection, kill criteria, or strategy-level reflection

### ISSUE THREAD

**Issue by rbeauchamp** [rbeauchamp/autoresearch#1]
## Problem

The current experiment loop in `program.md` is a flat optimize-and-iterate cycle:

```
LOOP FOREVER:
  1. Propose change
  2. Run experiment
  3. Keep or discard
  4. GOTO 1
```

This works well when the agent is making steady progress. But it has no mechanism to detect or escape **local optima** — when the current *approach* (not just the current *parameter*) has hit a ceiling.

In practice, an agent running this loop will:

1. Make initial progress on easy improvements (architecture, hyperparameters)
2. Hit diminishing returns as the easy wins are exhausted
3. Continue making **incremental tweaks** in the same subspace indefinitely — trying learning rate 0.003, then 0.004, then 0.0035 — never questioning whether the entire direction is productive
4. Degenerate into **brute-force random search** within whatever approach it started with

The loop runs "FOREVER" as instructed, but without metacognition it becomes a search algorithm, not a research process. A human researcher naturally pauses to ask "am I stuck? should I try something fundamentally different?" — the current prompt never triggers this behavior in the agent.

## Specific gaps

1. **No stall detection.** The loop doesn't track whether the target metric is actually improving over time. An agent can run 50+ experiments with zero improvement and keep going.

2. **No kill criteria.** The loop defines when to keep/discard individual experiments but never when to *abandon an approach*. Without pre-committed kill criteria, the agent cannot distinguish "this needs more tuning" from "this direction is exhausted."

3. **No strategy-level reflection.** The loop operates at one level: individual experiments. There's no outer loop that periodically asks "is this category of experiments working? should I pivot?"

4. **No forced divergence.** The loop never mandates trying something *qualitatively different*. An agent naturally gravitates toward small incremental changes (they're safer), which means it converges to a local optimum and stays there.

5. **No distinction between signal and noise.** The loop keeps/discards based on a single metric comparison. It doesn't help the agent distinguish "this approach reliably improves things" from "I got lucky once in 50 tries."

## Proposed fix

Add a lightweight **strategy checkpoint** mechanism to the experiment loop that:

- Tracks experiments-since-last-improvement as a running counter
- Triggers a mandatory strategy review after a configurable stall threshold
- Requires the agent to write a falsifiable hypothesis and kill criterion for each new strategy
- Forces qualitative pivots (not just parameter tweaks) after extended stalls

This should be minimal — a few paragraphs added to the loop section of `program.md` — keeping the spirit of the repo's simplicity while preventing the degenerate-to-random-search failure mode.
